pkg/controller/apimanager/apimanager_controller_test.go

```diff
@@ -74,22 +73,14 @@ func TestAPIManagerControllerCreate(t *testing.T) {
 		},
 	}
 
-	res, err := r.Reconcile(req)
-	if err != nil {
-		t.Fatalf("reconcile: (%v)", err)
-	}
-
-	if !res.Requeue {
-		t.Error("reconcile did not requeue request as expected. Requeuing due to setting of defaults should have been performed")
-	}
-
-	res, err = r.Reconcile(req)
-	if err != nil {
-		t.Fatalf("reconcile: (%v)", err)
-	}
+	endLoop := false
+	for i := 0; i < 100 && !endLoop; i++ {
+		res, err := r.Reconcile(req)
+		if err != nil {
+			t.Fatal(err)
+		}
 
-	if res.Requeue {
-		t.Error("reconcile did not finish end of reconciliation as expected. APIManager should have been reconciled at this point")
+		endLoop = !res.Requeue
 	}
```

Comment:
Should we check the value of endLoop after the loop? In the case endLoop is false after the loop it would mean a reconcile requeue has been received even if the contents we check in the test seem correct

Reply:
good catch.